resources/js/src/checkout/blocks/components/PaymentMethodContentRenderer.js

```diff
@@ -0,0 +1,105 @@
+import { MOLLIE_STORE_KEY } from '../store';
+import { createPaymentComponent } from './PaymentComponentFactory';
+
+/**
+ * Main Mollie Component - Orchestrates payment method rendering
+ * Handles common payment processing and delegates specific logic to child components
+ * @param {Object} props
+ */
+export const PaymentMethodContentRenderer = ( props ) => {
+	const { useEffect } = wp.element;
+	const { useSelect } = wp.data;
+
+	const {
+		activePaymentMethod,
+		billing,
+		item,
+		jQuery,
+		emitResponse,
+		eventRegistration,
+		requiredFields,
+		shippingData,
+		isPhoneFieldVisible,
+	} = props;
+
+	const { responseTypes } = emitResponse;
+	const { onPaymentSetup } = eventRegistration;
+
+	// Redux store selectors - only for payment processing
+	const selectedIssuer = useSelect(
+		( select ) => select( MOLLIE_STORE_KEY ).getSelectedIssuer(),
+		[]
+	);
+	const inputPhone = useSelect(
+		( select ) => select( MOLLIE_STORE_KEY ).getInputPhone(),
+		[]
+	);
+	const inputBirthdate = useSelect(
+		( select ) => select( MOLLIE_STORE_KEY ).getInputBirthdate(),
+		[]
+	);
+	const inputCompany = useSelect(
+		( select ) => select( MOLLIE_STORE_KEY ).getInputCompany(),
+		[]
+	);
+
+	const issuerKey =
+		'mollie-payments-for-woocommerce_issuer_' + activePaymentMethod;
+
+	// Main payment processing - stays centralized for all payment methods
+	useEffect( () => {
+		const onProcessingPayment = () => {
+			const data = {
+				payment_method: activePaymentMethod,
+				payment_method_title: item.title,
+				[ issuerKey ]: selectedIssuer,
+				billing_phone: inputPhone,
+				billing_company_billie: inputCompany,
+				billing_birthdate: inputBirthdate,
+				cardToken: '',
+			};
+			const tokenVal = jQuery( '.mollie-components > input' ).val();
```

Comment:
If this is all we need jQuery for, should we not think about doing it in vanilla JS instead?

A more thorough solution would be to expose this type of data through te redux store and then consume it via `withSelect` and/or a higher-order-component